content/_guides/sasl/emacs-erc.md

```diff
@@ -0,0 +1,156 @@
+
+;; A list of channels to auto-join
+(setq erc-autojoin-channels-alist
+	'(("irc.libera.chat" "#emacs" "#guile" "#scheme" "#erc" "#godot")))
+;; OR, if NONE..
+;; (setq erc-autojoin-channels-alist nil)
+
+;; from http://www.john2x.com/emacs.html
+;; Set the prompt to use the channel name.
+(setq erc-prompt  (lambda () (concat (buffer-name) "> ")))
+
+;; beautify the text on screen
+(setq erc-fill-function 'erc-fill-static)
+(setq erc-fill-static-center 10)
+
+;; my main function for connecting to libera.chat
+;; (see also https://libera.chat/guides/connect for regional hostnames)
+(defun my-erc-connect ()
+  (interactive)
+  (erc-tls :server "irc.libera.chat" :port 6697 :nick "MYNICK" :full-name "MY NAME"))
+(global-set-key (kbd "C-c e") 'my-erc-connect)
+
+;; Channel-tracking: Convenience functions for easily flipping between channels you are logged into
+;; from https://www.emacswiki.org/emacs/ErcChannelTracking#toc8
+;; (I don't use so much anymore, actually..)
+(defvar erc-channels-to-visit nil
+    "Channels that have not yet been visited by erc-next-channel-buffer")
+(defun erc-next-channel-buffer ()
+  "Switch to the next unvisited channel. See erc-channels-to-visit"
+  (interactive)
+  (when (null erc-channels-to-visit)
+    (setq erc-channels-to-visit 
+          (remove (current-buffer) (erc-channel-list nil))))
+  (let ((target (pop erc-channels-to-visit)))
+    (if target 
+        (switch-to-buffer target))))
+(global-set-key (kbd "C-c n") 'erc-next-channel-buffer)
+;; ======== ERC END ================================
+```
+3. In the my-erc-connect command you've defined, replace MYNICK with your nickname you plan to use (assuming no one else uses and has registered this nick).
+4. Either shut down your emacs and restart, or evaluate all the above lines.
+
+## B. Register
+
+If you have registered a nickname with libera already, and you remember the password, then skip to part C.
+Otherwise, register a nick with libera.chat, as follows:
+
+1. NOTE: In order to connect to irc.libera.chat without SASL (as you'll need to do here in order to register in the first place, right?) you can't be using a VPN. See [our notes on IP Range restrictions](guides/sasl#sasl-access-only-ip-ranges). Later, once you connect via SASL, below, you'll be able to use your VPN to connect.
+2. Connect to irc.libera.chat (unregistered), using the nick you plan to register. (You should be able to use the global key C-c n using the global-set-key you defined above to connect).
+3. Make sure the nick you plan to use is allowed. (libera's NickServ will give you an error if you are trying to use a nick that is already registered.)
+4. Register: Follow our instructions here to [register a nickname and password](/guides/registration) while signed in.
+5. Once your registration appears successful, then save your username/password to your password safe of preference (Unix *pass*, or *keepassxc*, etc.).
+
+## C. Install ERC-SASL functionality
```

Comment:
There should be an empty line after headings

```suggestion
## C. Install ERC-SASL functionality

```

Reply:
i can't tell the difference here .. (?) sorry